src/toil/fileStore.py

```diff
@@ -0,0 +1,1372 @@
+# Copyright (C) 2015-2016 Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, print_function
+
+import base64
+import cPickle
+import collections
+import errno
+import logging
+import os
+import shutil
+import stat
+import tempfile
+import time
+import uuid
+
+from contextlib import contextmanager
+from fcntl import flock, LOCK_EX, LOCK_UN
+from functools import partial
+from hashlib import sha1
+from Queue import Queue, Empty
+from threading import Thread, Semaphore, Event
+
+from toil.common import cacheDirName
+from toil.lib.bioio import makePublicDir
+
+logger = logging.getLogger( __name__ )
+
+
+class FileStore(object):
+    '''
+    Class used to manage temporary files, read and write files from the job store\
```

Comment:
Never mind. I'd rather copy the mistakes in this PR and fix them in a later one.